### ISSUE TITLE
Adding awkward vectors given in `pt`, `eta`, `phi`, `E` coordinates raises Error

### ISSUE THREAD

**Issue by nikoladze** [scikit-hep/vector#79]
For example

```pycon
>>> import awkward as ak
>>> import vector
>>> vector.__version__
'0.8.1'
>>> v = vector.awk(ak.Array([{'pt' : 1, 'eta' : 2, 'phi' : 3, 'E' : 4}]))
>>> v + v

[...]

~/.local/lib/python3.9/site-packages/vector/_compute/spatial/eta.py in rhophi_z(lib, rho, phi, z)
     41
     42 def rhophi_z(lib, rho, phi, z):
---> 43     return lib.nan_to_num(lib.arctanh(z / lib.sqrt(rho ** 2 + z ** 2)), nan=0.0)
     44
     45

<__array_function__ internals> in nan_to_num(*args, **kwargs)

TypeError: no implementation found for 'numpy.nan_to_num' on types that implement __array_function__: [<class 'awkward.highlevel.Array'>]

```

Probably because `nan_to_num` is not a ufunc ...

**Comment by jpivarski** [comment]
For this reason, `nan_to_num` and `isclose` were added to Awkward Array in [1.2.0rc4](https://github.com/scikit-hep/awkward-1.0/releases/tag/1.2.0rc4) (current release is 1.2.1 or 1.3.0rc3, depending on whether you use pre-releases).

I had thought that Vector checked `awkward.__version__` upon import, but I don't see any such check there. I'm thinking of other projects, such as this check in Awkward Array:

https://github.com/scikit-hep/awkward-1.0/blob/ee09c3ad6bf95944279e40ac92c428ea897fe8e1/src/awkward/_connect/_numba/__init__.py#L13-L36

(can't be enforced by `install_requires` because Numba for Awkward Array, like Awkward Array for Vector, is not a strict dependency.) I don't know, @henryiii, if you have opinions on this kind of thing, especially since you said that Python would be removing `distutils` in some version.

(I'm closing this issue because @nikoladze, your problem should be solved by a newer version of Awkward Array.)

**Comment by henryiii** [comment]
`packaging` is where the modern version handling sits. And you can/should use `importib.metadata` (backport: `importlib_metadata`) to check versions. We should also provide an awkward extra, so that `pip install vector[awkward]` would enforce the version.

**Comment by jpivarski** [comment]
I'll fix #78 if you set up the Awkward version check/extra package.  `:)`

**Comment by nikoladze** [comment]
Oh, sorry - forgot to upgrade awkward. I can confirm it works with `awkward-1.2.2`. Thanks!
